components/ui/common/SwipeableRow/SwipeableRow.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useMemo, useRef, useState } from "react";
+import { StyleSheet, View } from "react-native";
+import Animated, {
+  Easing,
+  SharedValue,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withTiming,
+} from "react-native-reanimated";
+import {
+  Gesture,
+  GestureDetector,
+  GestureStateChangeEvent,
+  PanGestureHandlerEventPayload,
+} from "react-native-gesture-handler";
+import { Handlers, SwipeableRowGestureContext } from "./types";
+import { SwipeableRowProvider } from "./SwipeableRowProvider";
+
+interface Props {
+  /**
+   * The left option to display when swiping left to right
+   */
+  leftOption?: JSX.Element;
+
+  /**
+   * The right option to display when swiping right to left
+   */
+  rightOption?: JSX.Element;
+
+  /**
+   * The shared value to use for the translateX value.
+   * This value will be updated while the user swipes
+   * as well as lets go.
+   */
+  translateX?: SharedValue<number>;
+
+  /**
+   * The component which will pan right and left as the user drags
+   */
+  children: React.ReactNode;
+}
+
+export const SwipeableRow = ({ leftOption, rightOption, children }: Props) => {
+  const [subscribers, setSubscribers] = useState<Handlers[]>([]);
+
+  const subscribe = useCallback((handlers: Handlers) => {
+    setSubscribers((subs) => [...subs, handlers]);
+
+    return () => {
+      setSubscribers((subs) => subs.filter((handler) => handler !== handlers));
+    };
+  }, []);
+
+  const swipeRightEnabled = Boolean(leftOption);
+  const swipeLeftEnabled = Boolean(rightOption);
+
+  const translateX = useSharedValue(0);
+  const subsTranslateX = useDerivedValue(() => translateX.value);
+
+  const context = useSharedValue<SwipeableRowGestureContext>({ startX: 0 });
+
+  const panGesture = useMemo(() => {
+    const broadcast = (
+      event: GestureStateChangeEvent<PanGestureHandlerEventPayload>,
+      name: keyof Handlers
+    ) => {
+      "worklet";
+      subscribers.forEach((handler) => {
+        handler[name]?.(event);
+      });
+    };
+
+    return Gesture.Pan()
```

Comment:
This is the new Gesture api in RNGH. The old one, which used the `PanGestureHandler` component and `useAnimatedGestureHandler` is apparently no longer going to be supported.